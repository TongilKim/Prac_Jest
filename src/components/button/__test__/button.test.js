import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../button';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom";


afterEach(cleanup);
it("renders Button component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
});


it("Check Button's label", () => {
    const { getByTestId } = render(<Button label="save"></Button>);
    // check if 'save' is included in the button label name
    expect(getByTestId('button')).toHaveTextContent("save");
})

it("matches snapshot", () => {
    const tree = renderer.create(<Button label="test"></Button>).toJSON();

    expect(tree).toMatchSnapshot();
})