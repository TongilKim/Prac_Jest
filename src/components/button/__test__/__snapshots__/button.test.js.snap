// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="btn-style"
  data-testid="button"
>
  test
</div>
`;
